### DISCUSSION TITLE
JSONPath: How to test unordered array equality?

### DISCUSSION THREAD

**Discussion by ens-scmeeu** [Orange-OpenSource/hurl#1903] [Q&A] [answered]
I'm trying to validate that the `name` key nested a few levels deep inside JSON objects (which is inside the array specified by the `objects` key) contains two items, but without respect to ordering.

This is what I tried to do, but it didn't parse (it can't seem to parse the JSON array - regardless I don't think this would do unordered compare anyway):

```hurl
jsonpath "$.objects[*].working.input.name" == ["foo", "bar"]
```

This works currently (but it is a brittle test if the ordering of the array changes):

```hurl
jsonpath "$.objects[0].working.input.name" == "foo"
jsonpath "$.objects[1].working.input.name" == "bar"
```

Any other way to do this compare in an order agnostic way?

**Comment by jcamiel** [comment, marked as answer]
Hi, 

`jsonpath "$.objects[*].working.input.name"`returns a collection so you could do this:

```
jsonpath "$.objects[*].working.input.name" includes "foo"
jsonpath "$.objects[*].working.input.name" includes "bar"
jsonpath "$.objects[*].working.input.name" count == 2
jsonpath "$.objects[*].working.input.name" isCollection
jsonpath "$.objects[*].working.input.name" nth 0 == "foo"
jsonpath "$.objects[*].working.input.name" nth 1 == "bar"
# etc....
```

(if I'm not mistaken ☺️)

**Reply by ens-scmeeu** [reply]
Awesome, thanks! I was just trying `contains` which didn't work, but `includes` as you specified works perfectly!

**Reply by jcamiel** [reply]
See #1896, we may merge `contains` and `includes` and just keep `contains`  because it's not obvious what's the difference between the two

**Reply by ens-scmeeu** [reply]
I think this is a really good idea (gave it a thumbs up). Admittedly I was scratching my head as to why there were both options that worked on different types.
